src/main/java/core/basesyntax/service/impl/StorageServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.StorageService;
+import core.basesyntax.strategy.CountStrategy;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+import java.util.Map;
+
+public class StorageServiceImpl implements StorageService {
+    private final OperationStrategy operationStrategy;
+    private final Storage storage = new Storage();
+
+    public StorageServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public Map<String, Integer> update(List<FruitTransaction> fruitTransactions) {
+        Map<String, Integer> fruitMap = storage.getFruitMap();
+        for (FruitTransaction transaction : fruitTransactions) {
+            FruitTransaction.Operation operation = transaction.getOperation();
+            CountStrategy countStrategy = operationStrategy.getCountStrategy(operation);
+            String fruitName = transaction.getFruit();
+            int operationQuantity = transaction.getQuantity();
+            int currentQuantity =
+                    operation == FruitTransaction.Operation.BALANCE ? 0 : fruitMap.get(fruitName);
```

Comment:
such logic should be moved to handlers of each operation

Reply:
I don't understand how to do this, because I have to give handlers the already existed current quantity.

Reply:
They can take it from storage itself. Handlers must have logic which will take value from db, than update it and save